packages/core/src/DataSchema.ts

```diff
@@ -0,0 +1,118 @@
+import { CircuitConfig } from '@unirep/circuits'
+
+export class DataSchema {
+    public schema: any[]
+    public config: CircuitConfig
+    public data: bigint[]
+
+    constructor(schema, config = CircuitConfig.default) {
+        this.config = config
+        this.schema = this.parseSchema(schema)
+        this.data = new Array(config.FIELD_COUNT).fill(BigInt(0))
+    }
+
+    parseSchema(schema: any[]): any[] {
+        let sumOffset = 0
+        let replOffset = 0
+        const maxSumOffset = 253 * this.config.SUM_FIELD_COUNT
+        const maxReplOffset =
+            253 * (this.config.FIELD_COUNT - this.config.SUM_FIELD_COUNT)
+
+        return schema.map((field, idx) => {
+            const { name, type, updateBy, ...extranousFields } = field
```

Comment:
"extraneous", this could also be called "extra"

Reply:
Done, using `extraFields` now